gui/default/syncthing/core/syncthingController.js

```diff
@@ -1603,20 +1603,23 @@ angular.module('syncthing.core')
                 // here as well...
                 $scope.devices = deviceMap($scope.config.devices);
 
-                $scope.saveConfig().then(function () {
-                    if (themeChanged) {
-                        document.location.reload(true);
-                    }
+                $('#settings').modal('hide').on('hidden.modal.bs', function () {
+                    $scope.saveConfig().then(function () {
+                        if (themeChanged) {
+                            document.location.reload(true);
+                        }
+                    });
                 });
+            } else {
+                $('#settings').modal('hide');
             }
-
-            $("#settings").off("hide.bs.modal").modal("hide");
```

Comment:
I don't quite understand this part. Is anyone able to explain why `$("#settings").off("hide.bs.modal").modal("hide")` is used here instead of a simple `$("#settings").modal("hide")`?